lib/resources/collection/index.js

```diff
@@ -571,7 +571,7 @@ Collection.prototype.save = function (ctx, fn) {
     if(errs) return done({errors: errs});
 
     // generate id before event listener
-    item.id = store.createUniqueIdentifier();
+    if (!item.id) item.id = store.createUniqueIdentifier();
```

Comment:
The `On post` event is called after this `item.id` is populated. What it does is simply prepopulate the id so it is available as `this.id` in the event. 

I think it will be possible to set `this.id` to something else, such as a 10 character alphanumeric id from this `On post` event without any other changes even in current master. Have you verified that you can't?